### PR TITLE
SPLAT-1175: enable vsphere02 as a target for vSphere CI jobs

### PR DESCRIPTION
**Pull request by rvanderp3**
The intention of this PR is to enable the transition of jobs from vSphere build01 to vSphere build02.  At present, all vSphere jobs are mapped to the current vSphere build01 cluster.

### CODE REVIEW

File: pkg/dispatcher/config.go [hunk 1]
```diff
@@ -114,6 +114,11 @@ func (config *Config) DetermineClusterForJob(jobBase prowconfig.JobBase, path st
 		return "", false, nil
 	}
 	if strings.Contains(jobBase.Name, "vsphere") && !isApplyConfigJob(jobBase) {
+		if cluster, ok := jobBase.Labels[api.ClusterLabel]; ok {
+			if cluster == string(api.ClusterVSphere02) {
+				return api.Cluster(cluster), false, nil
+			}
+		}
```
Comment (hongkailiu) [inline pkg/dispatcher/config.go:121]: Add a unit test for the case?

**Comment by rvanderp3** [comment]
/hold
placing hold until test is in place

**Pull request merged**